src/luaotfload-main.lua

```diff
@@ -44,31 +44,40 @@ luaotfload.fontloader_package     = "reference"    --- default: from current Con
 
 if not tex or not tex.luatexversion then
     error "this program must be run in TeX mode" --- or call tex.initialize() =)
-else
-    --- version check
-    local major    = tex.luatexversion / 100
-    local minor    = tex.luatexversion % 100
-    local revision = tex.luatexrevision --[[ : string ]]
-    local revno    = tonumber (revision)
-    local minimum  = luaotfload.min_luatex_version
-    local actual   = { major, minor, revno or 0 }
-    if actual [1] < minimum [1]
+end
+
+--- version check
+local major    = tex.luatexversion / 100
+local minor    = tex.luatexversion % 100
+local revision = tex.luatexrevision --[[ : string ]]
+local revno    = tonumber (revision)
+local minimum  = luaotfload.min_luatex_version
+local actual   = { major, minor, revno or 0 }
+if actual [1] < minimum [1]
     or actual == minimum and actual [2] < minimum [2]
     or actual == minimum and actual [2] == minimum [2] and actual [3] < minimum [3]
-    then
-        texio.write_nl ("term and log",
-                        string.format ("\tFATAL ERROR\n\z
-                                        \tLuaotfload requires a Luatex version >= %d.%d.%d.\n\z
-                                        \tPlease update your TeX distribution!\n\n",
-                                       (unpack or table.unpack) (minimum)))
-        error "version check failed"
-    end
+then
+    texio.write_nl ("term and log",
+                    string.format ("\tFATAL ERROR\n\z
+                                    \tLuaotfload requires a Luatex version >= %d.%d.%d.\n\z
+                                    \tPlease update your TeX distribution!\n\n",
+                                   (unpack or table.unpack) (minimum)))
+    error "version check failed"
+end
+
+if not utf8 then
+    texio.write_nl("term and log", string.format("\z
+        \tluaotfload: module utf8 is unavailable\n\z
+        \tutf8 requires Lua 5.3+; engine's _VERSION is '%s'\n\z
+        \n",
+        _VERSION))
+    error "module utf8 is unavailable"
 end
```

Comment:
Looks good to me. Given that we already (try to) check for a sufficiently new LuaTeX version, we might want to add a note why this could happen: The current message looks like a LuaTeX update would be required, but I think this should (once the other test is fixed) only trigger because of LuaJITTeX. There, updating wouldn't do much good.

Reply:
I added "Engines other than LuaTeX is not officially supported".